### PR TITLE
[stable/Jenkins] Added jenkins location configuration

### LINKED ISSUE #6513: [stable/jenkins] Jenkins URL is not set during deploy time
**Is this a request for help?**:
No
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
Feature Request

**Version of Helm and Kubernetes**:
Helm:
```
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.0", GitCommit:"f6025bb9ee7daf9fee0026541c90a6f557a3e0bc", GitTreeState:"clean"}
```

Kubernetes:
```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.2", GitCommit:"5fa2db2bd46ac79e5e00a4e6ed24191080aa463b", GitTreeState:"clean", BuildDate:"2018-01-18T10:09:24Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:05:37Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**:
stable/jenkins

**What happened**:
During deployment `JenkinsLocationConfiguration.xml` does not get filled out. If placing the instance behind an ingress you will need to go through another configuration step to get rid of a warning that crops up post-install.

**What you expected to happen**:
This configuration should get configured if you are configuring an ingress.

**How to reproduce it** (as minimally and precisely as possible):
1. Install the chart with ingress configuration enabled.
2. Log into the created Jenkins instance and observe the following error
```
JENKINS ROOT URL IS EMPTY BUT IS REQUIRED FOR THE PROPER OPERATION OF MANY JENKINS FEATURES LIKE EMAIL NOTIFICATIONS, PR STATUS UPDATE, AND ENVIRONMENT VARIABLES SUCH AS BUILD_URL.
PLEASE PROVIDE AN ACCURATE VALUE IN JENKINS CONFIGURATION.
```

**Anything else we need to know**:
I have a proposed fix for this issue and will be submitting a PR for review soon.

### PR DESCRIPTION
**Pull request by rbutcher**
**What this PR does / why we need it**:
Added jenkins.model.JenkinsLocationConfiguration.xml to config map to support setting JENKINS_URL to avoid warnings on first start and manual configuration.

**Which issue this PR fixes**:
fixes #6513

### CODE REVIEW

File: stable/jenkins/templates/config.yaml [hunk 1]
```diff
@@ -155,6 +155,20 @@ data:
       <pendingClasspathEntries/>
     </scriptApproval>
 {{- end }}
+  jenkins.model.JenkinsLocationConfiguration.xml: |-
+    <?xml version='1.1' encoding='UTF-8'?>
+    <jenkins.model.JenkinsLocationConfiguration>
+      <adminAddress>{{ default "nobody@nowhere.io" .Values.Master.JenkinsAdminEmail }}</adminAddress>
+{{- if .Values.Master.HostName }}
+{{- if .Values.Master.JenkinsUriSecure }}
```
Comment (jlegrone) [inline stable/jenkins/templates/config.yaml]: Where is this coming from? Could we use `Master.Ingress.TLS` instead?

File: stable/jenkins/templates/config.yaml [hunk 2]
```diff
@@ -155,6 +155,20 @@ data:
       <pendingClasspathEntries/>
     </scriptApproval>
 {{- end }}
+  jenkins.model.JenkinsLocationConfiguration.xml: |-
+    <?xml version='1.1' encoding='UTF-8'?>
+    <jenkins.model.JenkinsLocationConfiguration>
+      <adminAddress>{{ default "nobody@nowhere.io" .Values.Master.JenkinsAdminEmail }}</adminAddress>
```
Comment (jlegrone) [inline stable/jenkins/templates/config.yaml]: Please document `JenkinsAdminEmail` in the README.  Also, could we just default to an empty string?  Not sure who owns nowhere.io...

**Review by jlegrone — CHANGES_REQUESTED** [review]
Thanks for this!  It'll be nice to get rid of the warning.  Please see my comments and be sure to bump the chart version as well.

File: stable/jenkins/templates/config.yaml (hunk 1 above)
Comment (rbutcher) [reply stable/jenkins/templates/config.yaml]: You are totally right. There is no instance I can think of where I would not configure TLS on the ingress and want an https:// uri.

**Comment by rbutcher** [comment]
Its been 2 weeks with no activity on this. Is there something I need to do to get someone to look at this?

**Pull request merged**